timesource/timesource_test.go

```diff
@@ -271,20 +272,39 @@ func TestGetCurrentTimeInMillis(t *testing.T) {
 	}
 
 	expectedTime := uint64(11000)
-	n, e := GetCurrentTimeInMillis()
-	require.NoError(t, e)
+	n := GetCurrentTimeInMillis()
 	require.Equal(t, expectedTime, n)
 	// test repeat invoke GetCurrentTimeInMillis
-	n, e = GetCurrentTimeInMillis()
-	require.NoError(t, e)
+	n = GetCurrentTimeInMillis()
 	require.Equal(t, expectedTime, n)
-	e = Default().Stop()
+	e := Default().Stop()
 	require.NoError(t, e)
 
 	// test invoke after stop
-	n, e = GetCurrentTimeInMillis()
-	require.NoError(t, e)
+	n = GetCurrentTimeInMillis()
 	require.Equal(t, expectedTime, n)
 	e = Default().Stop()
 	require.NoError(t, e)
 }
+
+func TestGetCurrentTimeOffline(t *testing.T) {
+	// covers https://github.com/status-im/status-desktop/issues/12691
+	// panic: sync: negative WaitGroup counter
+	ntpTimeSourceCreator = func() *NTPTimeSource {
+		if ntpTimeSource != nil {
+			return ntpTimeSource
+		}
+		ntpTimeSource = &NTPTimeSource{
+			servers:           defaultServers,
+			allowedFailures:   DefaultMaxAllowedFailures,
+			fastNTPSyncPeriod: 1 * time.Millisecond,
+			slowNTPSyncPeriod: 1 * time.Second,
+			timeQuery: func(string, ntp.QueryOptions) (*ntp.Response, error) {
+				return nil, errors.New("offline")
+			},
+		}
+		return ntpTimeSource
+	}
+	_ = GetCurrentTime()
+	_ = GetCurrentTime()
```

Comment:
It's not that appearently we are expecting this invoking should return quickly, maybe add a timeout for the second invoking ?

Reply:
Not sure I understand the point, could you please elaborate?

PS. This was to test two consecutive calls to `GetCurrentTime` won't result in "panic: sync: negative WaitGroup counter" if offline.